### PR TITLE
Abstract a bit the uki commands

### PR DESCRIPTION
**Pull request by Itxaka**
Make use of the config with the runners and vfs so we can easily tests the actions being taken and substitute those structs for others

### CODE REVIEW

File: pkg/action/build-uki.go [hunk 1]
```diff
@@ -457,64 +456,25 @@ func copyFilesToImg(imgFile string, filesMap map[string][]string) error {
 	return nil
 }
 
-func GzipFile(sourcePath, targetPath string) error {
```
Comment (Itxaka) [inline pkg/action/build-uki.go:460]: moved to utils

File: pkg/action/build-uki.go [hunk 2]
```diff
@@ -457,64 +456,25 @@ func copyFilesToImg(imgFile string, filesMap map[string][]string) error {
...
-	defer gzipWriter.Close()
-
-	if _, err = io.Copy(gzipWriter, inputFile); err != nil {
-		return fmt.Errorf("error writing data to the compress initramfs file: %w", err)
-	}
-
-	return nil
-}
-
-func findKairosVersion(sourceDir string) (string, error) {
```
Comment (Itxaka) [inline pkg/action/build-uki.go:483]: same

File: pkg/action/build_uki_test.go [hunk 3]
```diff
@@ -0,0 +1,95 @@
...
+				switch cmd {
+				case "xorriso":
+					err := fs.WriteFile(filepath.Join(tmpDir, "elemental.iso"), []byte("profound thoughts"), constants.FilePerm)
+					return []byte{}, err
+				default:
+					return []byte{}, nil
+				}
+			}
+		})
+		It("Builds UKI Iso", func() {})
```
Comment (Itxaka) [inline pkg/action/build_uki_test.go:91]: nothing yet added

Comment (jimmykarily) [reply pkg/action/build_uki_test.go:91]: Happening in this PR or other? Thanks for writing this btw, I looked at the other test and I was too afraid to try this :D.

**Review by jimmykarily — APPROVED** [review]

**Comment by jimmykarily** [comment]
Nice job!

File: pkg/action/build_uki_test.go (hunk 3 above)
Comment (Itxaka) [reply pkg/action/build_uki_test.go:91]: Hopefully in this one lmao

**Pull request closed without merging**